src/dra/5D6C4.c

```diff
@@ -474,7 +474,158 @@ void AddHearts(s32 value) {
     }
 }
 
-INCLUDE_ASM("dra/nonmatchings/5D6C4", func_800FE97C);
+// Note: Arg3 is unused, but is given in the call from func_80113D7C
+s32 func_800FE97C(Unkstruct_800FE97C* arg0, s32 arg1, s32 arg2, s32 arg3) {
+    s32 ret;
+    s32 itemCount;
+
+    func_800F53A4();
+    arg0->unk0 = arg1 & ~0x1F;
+    arg0->unk4 = arg1 & 0x1F;
+    if (g_Status.defenseElement & arg0->unk0) {
+        arg2 *= 2;
+    }
+    if (g_Status.D_80097C2A & arg0->unk0) {
+        arg2 /= 2;
+    }
+    if (g_Status.D_80097C2C & arg0->unk0) {
+        if (!(g_Status.D_80097C2C & arg0->unk0 & 0x200)) {
+            return 0;
+        }
+        arg0->unk0 &= ~0x200;
+    }
+
+    if (g_Status.D_80097C2E & arg0->unk0) {
+        if (arg2 < 1) {
+            arg2 = 1;
+        }
+        arg0->unkC = arg2;
+        if (g_Status.hp != g_Status.hpMax) {
+            func_800FE8F0();
+            g_Status.hp += arg0->unkC;
+            if (g_Status.hpMax < g_Status.hp) {
+                g_Status.hp = g_Status.hpMax;
+            }
+        }
+        return 5;
+    }
+    // Player wearing cat-eye circlet. Same as above if-statement but
+    //  with arg2 doubled. Item description says "Big HP restore" so makes
+    //  sense
+    if (CheckEquipmentItemCount(ITEM_CAT_EYE_CIRCLET, HEAD_TYPE) != 0 &&
+        arg0->unk4 == 7) {
+        arg2 *= 2;
+        if (arg2 < 1) {
+            arg2 = 1;
+        }
+        arg0->unkC = arg2;
+        if (g_Status.hp != g_Status.hpMax) {
+            func_800FE8F0();
+            g_Status.hp += arg0->unkC;
+            if (g_Status.hpMax < g_Status.hp) {
+                g_Status.hp = g_Status.hpMax;
+            }
+        }
+        return 5;
+    }
+
+    // Ballroom mask???
```

Comment:
remove this comment since you already added the enum value?

Reply:
I want to call it out as a very unusual thing. The ballroom mask is not supposed to have any special properties and the code should not have to check if you are wearing two of them.

Reply:
i think It would be great if those implications are stated in the comment instead of just the question marks. Do you think you can do a brief multiline comment explanation?
/*
 * Explain unusual behaviour
 */

Reply:
Done, thank you for the feedback.